core/src/main/java/org/elasticsearch/index/mapper/MapperServiceModule.java

```diff
@@ -19,15 +19,48 @@
 
 package org.elasticsearch.index.mapper;
 
+import com.google.common.collect.ImmutableSet;
+
 import org.elasticsearch.common.inject.AbstractModule;
+import org.elasticsearch.common.util.ExtensionPoint;
+import org.elasticsearch.index.mapper.internal.AllFieldMapper;
+import org.elasticsearch.index.mapper.internal.FieldNamesFieldMapper;
+import org.elasticsearch.index.mapper.internal.IdFieldMapper;
+import org.elasticsearch.index.mapper.internal.IndexFieldMapper;
+import org.elasticsearch.index.mapper.internal.ParentFieldMapper;
+import org.elasticsearch.index.mapper.internal.RoutingFieldMapper;
+import org.elasticsearch.index.mapper.internal.SourceFieldMapper;
+import org.elasticsearch.index.mapper.internal.TTLFieldMapper;
+import org.elasticsearch.index.mapper.internal.TimestampFieldMapper;
+import org.elasticsearch.index.mapper.internal.TypeFieldMapper;
+import org.elasticsearch.index.mapper.internal.UidFieldMapper;
+import org.elasticsearch.index.mapper.internal.VersionFieldMapper;
+
+import java.util.Set;
 
 /**
- *
+ * The infrastructure around DocumentMappers and FieldMappers.
  */
 public class MapperServiceModule extends AbstractModule {
+    // TODO it should be possible to move all of the rootTypeParsers into rootParsers as "extensions" registered in Elasticsearch core.
+    private static final Set<String> ROOT_TYPE_PARSERS = ImmutableSet.of(IndexFieldMapper.NAME, SourceFieldMapper.NAME,
```

Comment:
Can we not use the old "root mapper" terminology? These are "metadata" fields now.